index.js

```diff
@@ -149,6 +149,45 @@ switch (process.platform) {
       }
     })
     break
+
+  case 'linux':
+    Object.assign(fsAdmin, {
+      clearAuthorizationCache () {
+        spawnSync('/bin/pkcheck', ['--revoke-temp'])
+      },
+
+      createWriteStream (filePath) {
+        // Prompt for credentials synchronously to avoid creating multiple simultaneous prompts.
+        if (!fsAdmin.testMode && spawnSync('/usr/bin/pkexec', ['/bin/dd']).status !== 0) {
```

Comment:
Is there a chance that `/usr/bin/pkexec` or `/bin/pkexec` could be missing from the user's machine? If so, I'd love for this code to fail in a predictable way (or revert to the default behaviour)

Reply:
Good catch. ⚡️  It is indeed possible that some distros won't include Polkit. In Atom, we are working around that by requiring `polkit-1` to be a dependency of the Atom .deb/.rpm package.

That said, I was thinking we wouldn't need to do anything in particular here because `spawnSync` will fail with a non-zero status code and a `Failed to obtain credentials` exception will be reported to users of this API. 

The rationale for that is that I am assuming `createWriteStream` would only be used when it is  already known that the current user does not have access to a certain file. Is that assumption correct? 💭

Reply:
The reason why I've asked is that `spawnSync` and similar APIs don't exit cleanly with a non-zero exit code in some cases, and I've had to try-catch for `ENOENT` explicitly to avoid it bubbling up to the user.

Here's an example using an interactive Node session (Ubuntu 19.04):

```
$ node
Welcome to Node.js v12.3.1.
Type ".help" for more information.
> const { spawnSync } = require('child_process')
undefined
> spawnSync('/bin/missing')
{
  error: Error: spawnSync /bin/missing ENOENT
      at Object.spawnSync (internal/child_process.js:1041:20)
      at spawnSync (child_process.js:606:24)
      at repl:1:1
      at Script.runInThisContext (vm.js:123:20)
      at REPLServer.defaultEval (repl.js:384:29)
      at bound (domain.js:415:14)
      at REPLServer.runBound [as eval] (domain.js:428:12)
      at REPLServer.onLine (repl.js:700:10)
      at REPLServer.emit (events.js:205:15)
      at REPLServer.EventEmitter.emit (domain.js:471:20) {
    errno: 'ENOENT',
    code: 'ENOENT',
    syscall: 'spawnSync /bin/missing',
    path: '/bin/missing',
    spawnargs: []
  },
  status: null,
  signal: null,
  output: null,
  pid: 33370,
  stdout: null,
  stderr: null
}
> 
```

Reply:
Oh interesting. Yeah I am relying on `status` being null in that circumstance, which will make the `status !== 0` conditional evaluate to true. Maybe there's a better way of structuring this?

Reply:
Maybe this could be extracted out to it's own function, rather than trying to inline it?

```diff
diff --git a/index.js b/index.js
index 67b04d6..001d74c 100644
--- a/index.js
+++ b/index.js
@@ -151,6 +151,16 @@ switch (process.platform) {
     break
 
   case 'linux':
+
+    function spawnSyncWithoutError(command, args) {
+      const result = spawnSync(command, args)
+      if (result.error !== null) {
+        return false
+      }
+
+      return result.status !== 0
+    }
+
     Object.assign(fsAdmin, {
       clearAuthorizationCache () {
         spawnSync('/bin/pkcheck', ['--revoke-temp'])
@@ -158,7 +168,7 @@ switch (process.platform) {
 
       createWriteStream (filePath) {
         // Prompt for credentials synchronously to avoid creating multiple simultaneous prompts.
-        if (!fsAdmin.testMode && spawnSync('/usr/bin/pkexec', ['/bin/dd']).status !== 0) {
+        if (!fsAdmin.testMode && spawnSyncWithoutError('/usr/bin/pkexec', ['/bin/dd'])) {
           const result = new EventEmitter()
           result.write = result.end = function () {}
           process.nextTick(() => result.emit('error', new Error('Failed to obtain credentials')))
```

I'm not sure what other cases you would be looking to cover here, but that would at least let you keep the overall flow easy-to-read...

Reply:
Cool, I refactored it a bit in a7429bd. I ended up not extracting a function, but I think it reads better now. Curious to hear your thoughts about it! 😊

Reply:
Works for me